### PR TITLE
Add rpc request size limits

### LINKED ISSUE #2078: Limit response sizes & and make error message a sequence of bytes
Spec update: https://github.com/ethereum/eth2.0-specs/pull/1837/files

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Add rpc request size limits

## Fixed Issue(s)

Fixes https://github.com/PegaSysEng/teku/issues/2078 (Does not include error message change)

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandler.java [hunk 1]
```diff
@@ -13,18 +13,25 @@
...
 import tech.pegasys.teku.storage.client.RecentChainData;
+import tech.pegasys.teku.util.config.Constants;
 
 public class BeaconBlocksByRootMessageHandler
     extends PeerRequiredLocalMessageHandler<BeaconBlocksByRootRequestMessage, SignedBeaconBlock> {
   private static final org.apache.logging.log4j.Logger LOG = LogManager.getLogger();
 
+  static final RpcException TOO_MANY_BLOCKS_REQUESTED =
+      new RpcException(INVALID_REQUEST_CODE, "Requested too many blocks");
+
```
Comment (Nashatyrev) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandler.java:34]: Singleton exception instances IMHO in not the right pattern (I also saw others and was going create an issue). Makes sense to change it to either static method or lambda
([stackoverflow question](https://stackoverflow.com/questions/15090664/is-it-safe-to-store-an-instance-of-an-exception-and-reuse-it/15090870))

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandler.java [hunk 2]
```diff
@@ -38,6 +45,10 @@ public void onIncomingMessage(
       final ResponseCallback<SignedBeaconBlock> callback) {
     LOG.trace(
         "Peer {} requested BeaconBlocks with roots: {}", peer.getId(), message.getBlockRoots());
+    if (message.getBlockRoots().size() > Constants.MAX_REQUEST_BLOCKS) {
```
Comment (Nashatyrev) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandler.java:48]: Is it possible situation? Shouldn't we fail on SSZ deserialization?
Nothing criminal of course, but looks strange to check here the condition which should be enforced implicitly by the `SSZList` max size.

**Review by Nashatyrev — APPROVED** [review]
LGTM

**Comment by cemozerr** [comment]
Closing this PR due to offline discussion where we decided to have a lower level check on request size limits.

**Pull request closed without merging**